harvard-huit/Harvard-Geospatial-Library
# Upgrade geoblacklight and suppress Blacklight deprecation errors

**Upgrade geoblacklight and suppress Blacklight deprecation errors.**
* * *

**JIRA Tickets**: 

- [HGL-480](https://jira.huit.harvard.edu/browse/HGL-480)
- [HGL-481](https://jira.huit.harvard.edu/browse/HGL-481)
- [HGL-482](https://jira.huit.harvard.edu/browse/HGL-482)

* Other Relevant Links (Mailing list discussion, related pull requests, etc.):

_The newest version of Geoblacklight (3.4.0) specifically fixes many of the deprecation issues (but many still remain):_
[Geoblacklight 3.4.0](https://github.com/geoblacklight/geoblacklight/releases/tag/v3.4.0)

_Here is a large blog post that goes into the deprecation warning issue in depth:_ 
[Blacklight 7.x, deprecation of view overrides, paths forward](https://bibwild.wordpress.com/2021/09/02/blacklight-7-x-deprecation-of-view-overrides-paths-forward/)

_There is a Gem installed in Blacklight, managed by Chris Beer (core maintaner) that allows you to set the logging level of the Blacklight warnings:_
[Deprecation Gem](https://github.com/cbeer/deprecation)

# What does this Pull Request do?
Rei reported that on every pageload of the site, several messages were saved in `/logs/gbl/geoblacklight/nginx/error.log` warning of deprecated code that will be removed in Blacklight 8. Because of the frequency of the warnings, the `error.log` file is getting to large and unmanageable.

Example: 
```
App 511 output: DEPRECATION WARNING: render_facet_partials is deprecated and will be removed from blacklight 8.0. (called from render_facet_partials at /usr/local/rvm/gems/ruby-2.6.6/gems/blacklight-7.19.1/app/helpers/blacklight/facets_helper_behavior.rb:39)
App 511 output: DEPRECATION WARNING: Calling render_facet_partials without passing the first argument (fields) is deprecated and will be removed in Blacklight 8.0.0. (called from _app_views_catalog__facets_html_erb__529338901624077753_69956412557280 at /home/app/webapp/app/views/catalog/_facets.html.erb:17)
```

- This PR upgrades Geoblacklight to 3.4.0 and Blacklight to 7.19.2 (along with all the dependencies that are part of those Gems). 
The upgrades needed additional changes to our custom code to display the map on the homepage. It also added new default sorting options, so I included them as well.
- Additionally, I fixed a few of our custom templates to remove deprecated code that wouldn't be picked up by future upgrades because we are overriding Blacklight / Geoblacklight views.
- There is also a new initializer in place that sets the Blacklight warnings to `silent`. Included in this initializer are all the other options, so that we can tweak this locally to test things, or to easily make changes in the future.  _This means we do not have to make any updates to Passenger to silence the warnings._
- Finally, per feedback from Bonnie, I have updated the alert message that appears after logging in with HarvardKey, removing the "from CAS" wording which could be confusing to the general user. 
  
# How should this be tested?

A description of what steps someone could take to:
* Fully rebuild the application (to install all the upgraded Gems): `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
* Confirm that the map on the homepage still functions correctly.
* Surf around the site. Perform a bunch of searches. Click on a bunch of detail pages. After doing so, look at your local `/log/nginx/error.log` file to confirm that you are not seeing any Blacklight deprecation warnings as you surf. (There should only be one, upon app startup).
* Confirm that the new search result sort options are working as expected.
* Confirm that adding / removing facets from search results are working as expected.
* **NOTE:** We cannot test the HarvardKey message until it has been deployed to Dev.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@dl-maura @mferrarini @enriquediaz @awoods
